kfp: Update choose-executor.md (#3525)
kubeflow/website: content/en/docs/components/pipelines/v1/installation/choose-executor.md

## Choosing the Workflow Executor

1. [Emissary executor](#emissary-executor) has been Kubeflow Pipelines' default executor since Feburay 2022 when KFP 1.8 went GA. 
   We recommend Emissary executor unless you have known compatibility issues with Emissary, in which case please submit your
   feedback in [the Emissary Executor feedback Github issue](https://github.com/kubeflow/pipelines/issues/6249).

1. [Docker executor](#docker-executor) is avaiable as a legacy choice. In case you do have compatibilty issues with Emissary executor,

>    and your cluster is running on an older version of Kubernests (<1.20), you can configure to use Docker executor.

Reviewer: ```suggestion
   and your cluster is running on an older version of Kubernetes (<1.20), you can configure to use Docker executor.
```

Note that Argo Workflows support other workflow executors, but the Kubeflow Pipelines
team only recommend choosing between emissary executor and docker executor.

### Emissary Executor

> Emissary executor is a relatively new workflow executor. It was first released in Argo Workflows v3.1 (June 2021).

Reviewer: ```suggestion
Emissary executor is the **default** workflow executor for Kubeflow Pipelines v1.8+. It was first released in Argo Workflows v3.1 (June 2021).
```

The Kubeflow Pipelines team believe that its architectural and portability
improvements can make it the default executor that most people should use going forward.

* Container Runtime: any

> * Reliability: not yet well-tested and not yet popular, but the Kubeflow Pipelines

Reviewer 2: We can probably update this sentence to `Well tested and it is the default Argo executor from >= v3.3.`. But feel free to merge the current change without update, since our main goal is to call out the default executor.

Therefore, these components will continue to work on emissary executor without
modifications.

### Docker Executor

> Docker executor used to be the **default** workflow executor before Kubeflow Pipelines v1.8.

Reviewer: ```suggestion
Docker executor used to be the default workflow executor before Kubeflow Pipelines v1.8.
```
